crates/chia-tools/src/bin/validate-blockchain-db.rs

```diff
@@ -183,6 +183,54 @@ features that are validated:
             let amount = u64::from_be_bytes(row.get::<_, [u8; 8]>(4).expect("missing amount"));
             additions.insert(coin_name, (ph, parent, amount, reward));
         }
+
+        // first ensure that the reward coins for this block are all included in
+        // the coin record table.
+        let rewards = block.get_included_reward_coins();
+        for add in &rewards {
+            let new_coin_id = add.coin_id();
+            let Some((ph, _parent, amount, coin_base)) = additions.get(new_coin_id.as_slice())
+            else {
+                panic!("at height {height} the block created a reward coin {new_coin_id} that's not in the coin_record table");
+            };
+            // TODO: ensure the parent coin ID is set correctly
+            assert_eq!(ph, add.puzzle_hash.as_slice(),
+                "at height {height} the reward coin {new_coin_id} has an incorrect puzzle hash in the coin_record table {} expected {}",
+                hex::encode(ph),
+                add.puzzle_hash
+            );
+            // ensure the parent hash has the expected look
+            assert_eq!(*amount, add.amount, "at height {height} reward coin {new_coin_id} has amount {} in coin_record table, but the block has amount {}", amount, add.amount);
+            // this is a reward coin
+            assert!(coin_base, "at height {height} the reward coin {new_coin_id} is not marked as coin-base in the database");
```

Comment:
This doesn't look right, IIUC we're checking a reference to bool, not bool itself. Applying the suggestion above would make this comparison againt the bool directly.